### PR TITLE
Report screen

### LINKED ISSUE #371: Report details screen
* a admin should be able to see the reports of the crew/ pilot
* (with a report parametter if too much conflict with other issues)

### PR DESCRIPTION
**Pull request by hacknstack**
Closes #371
issues to fix (other issues):
- flightTrace doesn't work 
- PilotReport are printed as normal Report

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt [hunk 1]
```diff
@@ -45,15 +45,24 @@ class FinishedFlightsViewModel(val repository: Repository, val userId: String) :
...
   val flightReports = _flightReports.asStateFlow()
+  val flightReportsUsers = _flightReportsUsers.asStateFlow()
 
   fun refresh() {
     refreshUserAndFlights()
   }
 
+  fun getUserWithId(id: String) {
+    viewModelScope.launch {
+      _currentUser.value = repository.userTable.get(id ?: UNSET_ID, onError = { onError(it) })
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt:61]: `id ?: UNSET_ID` is unecessary and doesn't work anyway. Just use `id`

File: app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt [hunk 2]
```diff
@@ -105,6 +114,10 @@ class FinishedFlightsViewModel(val repository: Repository, val userId: String) :
       viewModelScope.launch {
         _flightReports.value =
             repository.reportTable.retrieveReports(flightId, onError = { onError(it) })
+        _flightReportsUsers.value =
+            _flightReports.value!!.map { report ->
+              repository.userTable.get(report.author ?: UNSET_ID, onError = { onError(it) })!!
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt:119]: Same here with `report.author ?: UNSET_ID`

File: app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+@Composable
+fun ReportDetailsScreen(
+    flightId: String,
+    finishedFlightsViewModel: FinishedFlightsViewModel,
+    isAdmin: Boolean,
+    userId: String,
+    navController: NavHostController
+) {
+  val flight by finishedFlightsViewModel.getFlight(flightId).collectAsStateWithLifecycle()
+  val reportId by finishedFlightsViewModel.flightReports.collectAsStateWithLifecycle()
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt:31]: Nitpick: should be `reportIds` as it is a list

**Review by Prolas — APPROVED** [review]
LGTM

File: app/src/main/java/ch/epfl/skysync/screens/reports/CrewReport.kt [hunk 4]
```diff
@@ -142,6 +142,7 @@ fun CrewReportScreen(
               if (!hasError(errorLittleChampagne, errorBigChampagne, errorPrestigeChampagne)) {
                 val report =
                     CrewReport(
+                        // todo setup flight id
```
[the remark this answers is not in the document]
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/screens/reports/CrewReport.kt]: fixed

File: app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt (hunk 3 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt]: fixed

File: app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt (hunk 2 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt]: fixed

File: app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt (hunk 1 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt]: fixed

**Pull request merged**